File: init/fast/nightside/precip_shape.m
function Q = precip_shape(pg, Qpeak, Qbackground)
%% makes a 2D Gaussian shape in Latitude, Longitude
arguments
  pg (1,1) struct
  Qpeak (1,1) {mustBeNonnegative,mustBeFinite}
  Qbackground (1,1) {mustBeNonnegative,mustBeFinite}
end

displace = 10 * pg.mlat_sigma;

mlatctr = pg.mlat_mean + displace * tanh((pg.MLON - pg.mlon_mean) / (2*pg.mlon_sigma) );
% changed so the arc is wider compared to its twisting

S = exp(-(pg.MLON - pg.mlon_mean).^2/2/pg.mlon_sigma^2) .* ...
    exp(-(pg.MLAT - mlatctr - 1.5*pg.mlat_sigma).^2/2/pg.mlat_sigma^2);
Q = Qpeak * S;

Q(Q < Qbackground) = Qbackground;

end % function

File: init/fast/nightside/precip_shape.py
import numpy as np
import xarray


def precip_shape(pg: xarray.Dataset, Qpeak: float, Qbackground: float) -> np.ndarray:
    """
    makes a 2D Gaussian shape in Latitude, Longitude
    """

    mlon_mean = pg.mlon.mean().item()
    mlat_mean = pg.mlat.mean().item()

    displace = 10 * pg.mlat_sigma

    mlatctr = mlat_mean + displace * np.tanh((pg.mlon.data - mlon_mean) / (2 * pg.mlon_sigma))
    # changed so the arc is wider compared to its twisting

    S = np.exp(-((pg.mlon.data - mlon_mean) ** 2) / 2 / pg.mlon_sigma ** 2) * np.exp(
        -((pg.mlat.data - mlatctr - 1.5 * pg.mlat_sigma) ** 2) / 2 / pg.mlat_sigma ** 2
    )
    Q = Qpeak * S

    Q[Q < Qbackground] = Qbackground

    return Q
